algorithms/unit_tests/TestStdAlgorithmsCompileOnly.cpp

```diff
@@ -507,6 +507,24 @@ struct TestStruct {
   }
 };
 
+#ifndef KOKKOS_ENABLE_CXX17
+template <typename ViewType>
+constexpr bool
+test_kokkos_iterator_satify_std_random_access_iterator_concept() {
+  return std::random_access_iterator<
+      Kokkos::Experimental::Impl::RandomAccessIterator<ViewType>>;
+}
+
+static_assert(test_kokkos_iterator_satify_std_random_access_iterator_concept<
+              Kokkos::View<int *>>());
+static_assert(test_kokkos_iterator_satify_std_random_access_iterator_concept<
+              Kokkos::View<const int *>>());
+static_assert(test_kokkos_iterator_satify_std_random_access_iterator_concept<
+              const Kokkos::View<int *>>());
+static_assert(test_kokkos_iterator_satify_std_random_access_iterator_concept<
+              const Kokkos::View<const int *>>());
```

Comment:
Why do we do const-qualified views as template params?

Reply:
It just looked like a gap in support.

Reply:
How so? It does look like a conscious decision when we did implement it as a specialization.
I fail to see what use case this supports.

The way I understand it is that we said we have `[c]{begin, end}` free functions that take a Kokkos view and return some sort of random iterator whose type is not specified.
Ensuring that the standard library random iterator concept is modeled by that this undisclosed type looks worthwhile to me.
But I am reluctant to approve that change that lets the view template argument have cv-qualifiers.  If anything I like the status quo better.

Reply:
Then let's just postpone `const` support (possibly until we decide to move the class out of `Impl`).

Reply:
Here you go: https://github.com/kokkos/kokkos/pull/7451/commits/e1d3d2d2dfd5a320931abc47d222928d2851cb00